packages/htmlbars-compiler/tests/html-compiler-test.js

```diff
@@ -199,12 +212,22 @@ function shouldBeVoid(tagName) {
   var div = document.createElement("div");
   div.appendChild(fragment.cloneNode(true));
 
-  var tag = '<' + tagName + ' data-foo="bar">';
-  var closing = '</' + tagName + '>';
+  // IE8 adds a ':' before some tags when cloned
+  // <keygen> becomes <:keygen>
+  var tag;
+  var closing;
+  if (div.innerHTML.indexOf(':') >= 0) {
+    tag = '<:' + tagName + ' data-foo="bar">';
+    closing = '</:' + tagName + '>';
+  } else {
+    tag = '<' + tagName + ' data-foo="bar">';
+    closing = '</' + tagName + '>';
+  }
+    
   var extra = "<p>hello</p>";
```

Comment:
Please implement the fix for this problem in `normalizeInnerHTML` in the `ie8InnerHTML` clause.  Should be doable with a regex search like the other fixes there.